rnd/autogpt_builder/src/components/CustomEdge.tsx

```diff
@@ -25,12 +33,47 @@ const CustomEdgeFC: FC<EdgeProps<CustomEdgeData>> = ({ data, selected, source, s
     `M ${sourceX - 5} ${sourceY} C ${sourceX + 128} ${sourceY - yDifference - 128} ${targetX - 128} ${sourceY - yDifference - 128} ${targetX + 3}, ${targetY}` :
     path;
 
+  console.table({ id, sourceX, sourceY, targetX, targetY, sourcePosition, targetPosition, path, labelX, labelY });
+
   return (
-    <BaseEdge
-      style={{ strokeWidth: 2, stroke: (data?.edgeColor ?? '#555555') + (selected ? '' : '80') }}
-      path={edgePath}
-      markerEnd={markerEnd}
-    />
+    <>
+      <BaseEdge
+        path={edgePath}
+        markerEnd={markerEnd}
+        style={{
+          strokeWidth: isHovered ? 3 : 2,
+          stroke: (data?.edgeColor ?? '#555555') + (selected || isHovered ? '' : '80')
+        }}
+      />
+      <path
+        d={edgePath}
+        fill="none"
+        strokeOpacity={0}
+        strokeWidth={20}
+        className="react-flow__edge-interaction"
+        onMouseEnter={() => setIsHovered(true)}
+        onMouseLeave={() => setIsHovered(false)}
+      />
+      <EdgeLabelRenderer>
+        <div
+          style={{
+            position: 'absolute',
+            transform: `translate(-50%, -50%) translate(${labelX}px,${labelY}px)`,
+            pointerEvents: 'all',
+          }}
+          className="edge-label-renderer"
+        >
+          <button
+            onMouseEnter={() => setIsHovered(true)}
+            onMouseLeave={() => setIsHovered(false)}
+            className={`edge-label-button ${isHovered ? 'visible' : ''}`}
+            onClick={onEdgeClick}
+          >
+            <X size={14} />
```

Comment:
When working with lucide's icon's and displaying them on the FE  try leveraging classNames with tailwind. Using the size class from tailwind will set height and width.

```suggestion
            <X className="size-4" />
```

Reply:
thats cool!